src/oop/programs_with_multiple_classes/medium/q10/MainTest.java

```diff
@@ -0,0 +1,131 @@
+    private final String varSize = "size";
+    private final String varColour = "colour";
+    private final String varVegtables = "vegetables";
+    private final String methodName = "eating";
+    public ObjectTest vegetable;
+    public ObjectTest cupboard;
+
+    @BeforeEach
+    public void setup() throws Throwable {
+        String vegetableClassString = "oop.programs_with_multiple_classes.medium.q10." + classVegtable;
+        String cupboardClassString = "oop.programs_with_multiple_classes.medium.q10." + classCupboard;
+        vegetable = new ObjectTest(vegetableClassString);
+        cupboard = new ObjectTest(cupboardClassString);
+        String modifiedVegtableMessage =
+                "You have modified the provided portions of class " + classVegtable + ". Please revert them to the original state.";
+        assertTrue(vegetable.hasField(varSpoiled, boolean.class), modifiedVegtableMessage);
+        assertTrue(vegetable.hasModifier(varSpoiled, "private"), modifiedVegtableMessage);
+        assertTrue(vegetable.hasField(varWeight, double.class), modifiedVegtableMessage);
+        assertTrue(vegetable.hasModifier(varWeight, "private"), modifiedVegtableMessage);
+        assertTrue(vegetable.hasField(varOrigin, String.class), modifiedVegtableMessage);
+        assertTrue(vegetable.hasModifier(varOrigin, "private"), modifiedVegtableMessage);
+        String modifiedCupMessage =
+                "You have modified the provided portions of class " + classCupboard + ". Please revert them to the original state.";
+        Class<?>[] classArguments = {int.class, String.class};
+        assertTrue(cupboard.hasConstructor(classArguments), modifiedCupMessage);
+        assertTrue(cupboard.hasModifier(classArguments, "public"), modifiedCupMessage);
+
+        Object[][] test = new Object[][]{{2, "Red"}, {10, "Black"}, {30, "Green"}};
+        for (int num = 0; num < test.length; num++) {
+            int size = (int) test[num][0];
+            String colour = (String) test[num][1];
+            Object[][] arguments = {
+                    {size, int.class},
+                    {colour, String.class}
+            };
+            Object tableInstance = cupboard.createInstance(arguments);
+            _assertEquals(size, cupboard.getFieldValue(tableInstance, varSize), modifiedCupMessage);
+            _assertEquals(colour, cupboard.getFieldValue(tableInstance, varColour), modifiedCupMessage);
+            _assertArrayEquals(new Vegetable[size], cupboard.getFieldValue(tableInstance, varVegtables), modifiedCupMessage);
+        }
+    }
+
+    @Test
+    public void cupboardClassHasCorrectAttributes() {
+        assertTrue(cupboard.hasField(varSize, int.class),
+                "Your " + classCupboard + " class is missing the " + varSize + " attribute.");
+        assertTrue(cupboard.hasModifier(varSize, "private"),
+                "Your " + classCupboard + " class " + varSize + " attributes does not have the correct visibility modifier.");
```

Comment:
Can we change this to "The x attribute in your y class does not have the correct visibility modifier" ?